### ISSUE TITLE
Support for input:focus pseudo-element selector

### ISSUE THREAD

**Issue by josemigallas** [airbnb/enzyme#1709]
Following up on #703 I would like to request this one, so that it can be checked if an input has focus:
```
const wrapper = mount('<MyComponent />'); // Suppose this has an input inside.
const input = wrapper.find('input');
expect(input.is(':focus')).toBe(true);
```
Right now an error is thrown:
> Enzyme::Selector does not support pseudo-element or pseudo-class selectors.

I'm using `"enzyme": "^3.3.0"`.

**Comment by ljharb** [comment]
This is a tricky one - specifically, it wouldn't ever work in `shallow`, only in `mount`, and only if the DOM implementation allowed this to be detected (i think some browsers don't, and I'm not sure if jsdom does or not).

**Comment by ljharb** [comment]
@chenesan please do! note, though, that it'd have to work in jsdom 6+, because jsdom has aggressively dropped support of older environments in newer versions.

**Comment by ljharb** [comment]
Fixed in #1965
